packages/core/src/map.ts

```diff
@@ -0,0 +1,69 @@
+export class Map<K, V> {
+    private map: { key: K; value: V }[] = []
+
+    size(): number {
+        return this.map.length
+    }
+
+    set(key: K, value: V): void {
+        const entry = this.getEntryByKey(key)
+        if (entry) {
+            entry.value = value
+        } else {
+            this.map.push({ key, value })
+        }
+    }
+
+    get(key: K): V | undefined {
+        const entry = this.getEntryByKey(key)
+        return entry ? entry.value : undefined
+    }
+
+    has(key: K): boolean {
+        return this.getEntryByKey(key) !== undefined
+    }
+
+    delete(key: K): boolean {
+        const entryIndex = this.findIndexByKey(key)
```

Comment:
you can use DeviceScript-specific `Array.insert()` method, specifically `this.entries.insert(entryIndex, -1)`